Launchers: rework Static Quicklists/Actions (#98)
elementary/docs: apis/launchers.md

As you can see, the method `set_progress` takes a `double` value type and is a range between `0` and `1`: from 0% to 100%. As with badges, Don't forget that `set_progress_visible` must be `true` and `.begin` is required for asynchronous methods.

## Actions

Actions are specific functions your app can perform without already being open; think of them as alternate and more specific entry points into your app. Actions appear in the context menu of your app icon in the Applications Menu and Dock, and are searchable by name from the Applications Menu.

Adding actions to a `.desktop` file does not involve writing any code or using any external dependencies, though your app needs a way to distinguish between actions, e.g. with command line flags.

> To add an action, add it by unique ID to a new `Actions` line in your app's .desktop file: 

Reviewer: "Add it by unique ID" doesn't sound natural to me either. Maybe something like:

```suggestion
Actions must first be declared in a new `Actions` line in your app's .desktop file. This line should contain a `;` separated list of unique action names:
```

```ini

> Actions=ActionID;

Reviewer: I'm not sure this part is super clear about where the `Actions` line should go. It would probably help to imply more of the desktop file:

```suggestion
[Desktop Entry]
Name=Hello Again
[...]
Actions=ActionID;
```

Author: yeah good call

```

Then below, add the action itself using the same unique ID:

```ini
[Desktop Action ActionID]
Name=The name of the action

> Icon=An icon for the action (optional)
> Exec=application executable and any command line arguments
> ```

Reviewer: I don't love mixing "code" and explanation. Perhaps we should do something like:
```suggestion
Icon=com.github.myteam.myapp.action-id
Exec=com.github.myteam.myapp --action-id
```

The `Icon` line is optional and should be an icon which represents the action that will be performed. The `Exec` line is required and should be your app's executable name and any command line argument required to trigger the action.
```

Reviewer: Ah formatting got messed up here with the suggestion since it contains "```". So I guess don't try to directly commit this suggestion :sweat_smile:

> The action name should not include your app's name, as it will always be displayed alongside your app. The action icon should also not be your app icon, as it may be shown in the menu for your app icon, or badged on top of the app icon.

Reviewer: perhaps wrap this is a hint block?
```suggestion
{% hint style="info" %}
The action name should not include your app's name, as it will always be displayed alongside your app. The action icon should also not be your app icon, as it may be shown in the menu for your app icon, or badged on top of the app icon.
{% endhint %}
```
